pkg/interface/src/logic/lib/util.js

```diff
@@ -325,3 +326,12 @@ export function numToUd(num) {
     f.join('.')
   )(num.toString())
 }
+
+export function usePreventWindowUnload(shouldPreventDefault) {
+  React.useEffect(() => {
+    if (!shouldPreventDefault) return;
+    const handleBeforeUnload = event => event.preventDefault();
```

Comment:
This doesn't work on Chrome for whatever reason, you can still close the tab.

Reply:
Looks like Chrome uses a slightly different approach. Give this a shot.